src/main/kotlin/no/digdir/servicecatalog/service/PublicServiceService.kt

```diff
@@ -89,4 +91,29 @@ class PublicServiceService(private val publicServiceRepository: PublicServiceRep
         findPublicServiceById(id, catalogId)
             ?.takeIf { it.published }
             ?: throw CustomNotFoundException()
+
+    private fun getAllCatalogIds(): List<String> {
+        return mongoOperations
+            .query(no.digdir.servicecatalog.model.Service::class.java)
+            .distinct("catalogId")
+            .`as`(String::class.java)
+            .all()
+    }
```

Comment:
vi tjener lite på å være så fancy, kan like så greit gjøre noe ala dette:
```suggestion
    private fun getAllCatalogIds(): List<String> {
        return publicServiceRepository.findAll().map { it.catalogId }
    }
```